0022-lnd-plan-of-attack.adoc

```diff
@@ -0,0 +1,318 @@
+For this current one, we will make the following assumptions (which will influence the design of the new messaging):
+
+- Adding support for the HALight protocol can be done by _adding_ libp2p protocols instead of modifying existing ones (as it is currently the case)
+
+This should allow us to finish this spike and once the new communication protocol is defined, add support for HALight to it based on the needs that will be revealed in this spike.
+
+==== Lightning is not a public ledger
+
+At the moment, the implementation of the rfc003 protocol is based on the `HtlcEvents` trait.
+Said trait exposes the events that can happen as part of the protocol execution.
+
+The implementations of that make use of the `btsieve` module which cannot be used for Lightning as already mentioned above.
+Hence, the integration of Lightning will likely require adding an "Events" traits, conceptually similar to `HtlcEvents`.
+`HtlcEvents` is already specific to a single ledger, so we can simply add a trait `halight::Events` that describes all the events that can happen during the execution of the HALight protocol.
+
+Note: `HtlcEvents` will be decomposed into single function traits with https://github.com/comit-network/comit-rs/issues/1779[#1779].
+The design of `halight::Events` should follow accordingly.
+
+With Lightning being a layer-2 protocol, there is the chance that during a swap, there is a chance that a channel gets closed in the middle of a swap.
+For our usecase, we assume that this is safely handled by LND and we get notified accordingly of whether the payment has been successful or not.
+There is no plan to monitor the underlying blockchain for this.
+
+==== Standard LN does not support payments without the other party knowing the pre-image
+
+There is an experimental feature called "hold-invoice" which we can utilize to design a symmetric protocol.
+As a consequence, users will have to compile LND with the feature flag "invoicesrpc" to get the desired functionality.
+
+==== Our current REST API endpoints are not capable of expressing the concept of a protocol per ledger
+
+At the moment, our REST API only exposes the following endpoints for submitting swaps: `/swaps/rfc003`.
+All the information is encoded in the HTTP body, which not only complicates the deserialization process but also makes it hard for clients to construct a correct payload.
+
+We can kill two birds with one stone if we choose to not add support for the HALight protocol on the existing endpoint but instead add new ones.
+
+. adding new routes is automatically backwards compatible and downstream projects can adapt them gradually
+. the new routes can take a different, less complicated design which aids error messages, deserialization code and client side implementations
+
+The proposal is roughly outlined here: https://github.com/comit-network/comit-rs/issues/1326.
+
+To start with, I would add the following routes: (bitcoin to LN seems pointless but of course we can still add it)
```

Comment:
Just because there are business offerings doesn't mean it is not pointless :)

Just trying to reduce the scope here but we can ofc add any combination.

Reply:
BTC to LN is also not cross-chain. We have this discussion at some point and I remember we agreed that it is not a priority for us?